### PR TITLE
std.math: Remove deprecated creal functions

### PR DESCRIPTION
**Pull request by ibuclaw**
First deprecation was in Jan 2018.

Unittests have been copied across to std.complex where appropirate.

### CODE REVIEW

**Comment by ibuclaw** [comment]
Depends on dlang/dmd#11123

**Comment by ibuclaw** [comment]
```
../dmd/src/dmd/doc.d(368,34): Error: file `"default_ddoc_theme.ddoc"` cannot be found or not in a path specified with `-J`
.generated/stable_dmd-2.088.0/dmd2/linux/bin64/dmd failed with exit code 1.
posix.mak:928: recipe for target '.generated/ddoc_preprocessor' failed
```
Did someone in the dmd repo break the ddoc build?

**Comment by ibuclaw** [comment]
Freebsd/32 is green, so knock wood the rest should follow...

**Comment by thewilsonator** [comment]
Buildkite looks unrelated.

File: std/math.d [hunk 1]
```diff
@@ -668,37 +628,6 @@ deprecated
...
-    import std.meta : AliasSeq;
-    static foreach (T; AliasSeq!(cfloat, cdouble, creal))
-    {{
-        T f = -12+3i;
-        assert(abs(f) == hypot(f.re, f.im));
-        assert(abs(-f) == hypot(f.re, f.im));
-    }}
-}
-
-import std.meta : AliasSeq;
```
Comment (Geod24) [inline std/math.d:684]: Why was this removed ? It's a breaking change.

Comment (dnadlinger) [reply std/math.d:684]: I'm having flashbacks here – didn't we fix selective imports to be private a few years ago?

Comment (dnadlinger) [reply std/math.d:684]: (i.e. the infamous "bug 314")

Comment (ibuclaw) [reply std/math.d:684]: As @dnadlinger said, did not anticipate there being a `public:` being in effect here.

Comment (ibuclaw) [reply std/math.d:684]: https://issues.dlang.org/show_bug.cgi?id=20836
